File: SpringBootScala/src/main/scala/com/tim/scala/services/UserService.scala
package com.tim.scala.services

import com.tim.scala.models.User
import com.tim.scala.repository.UserRepository
import javax.transaction.Transactional
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.data.jpa.repository.Modifying
import org.springframework.stereotype.Service

@Service
class UserService extends IUserService {

  @Autowired
  private val userRepository: UserRepository = null

  @Transactional
  def getAllUsers: java.util.List[User] = userRepository.findAll()

  @Transactional
  def getUserById(id: Long): User =  userRepository.getOne(id)

  @Modifying(flushAutomatically = true)
  def saveUser(user: User) = userRepository.save(user)

  @Modifying(flushAutomatically = true)
  def deleteUser(user: User) = userRepository.delete(user)

  @Modifying(flushAutomatically = true)
  def deleteUserById(id: Long) = userRepository.deleteById(id)

}

File: webshop/src/main/java/com/tim/webshop/services/UserService.java
package com.tim.webshop.services;


import com.tim.webshop.models.Users;

import java.util.List;

public interface UserService {

    Users save(Users user);
    List<Users> findAll();
    Users findOne(String username);

    <S extends Users> S saveAndFlush(S entity);

    Users findByUsername(String username);

    boolean existsByEmail(String email);
}

